File: api-pytest-framework/features/steps/booking_delete_steps.py
"""Steps BDD pour booking_delete.feature -- DELETE /booking/{id}."""
from pytest_bdd import when, then
from pages.booking_page import BookingPage


# ── When : DELETE /booking/{id} ───────────────────────────────────────────────

@when("j'envoie DELETE /booking/{id} avec mon token (Cookie)")
def delete_booking_valid(ctx):
    ctx["response"] = ctx["booking"].delete_booking(ctx["booking_id"])


@when("j'envoie GET /booking/{id} apres la suppression")
def get_booking_after_delete(ctx):
    ctx["response"] = ctx["booking"].get_booking(ctx["booking_id"])


@when("j'envoie DELETE /booking/{id} sans header d'authentification")
def delete_no_token(ctx):
    resp_create = ctx["booking"].create_booking()
    ctx["booking_id"] = resp_create.json()["bookingid"]
    ctx["response"] = BookingPage().delete_booking(ctx["booking_id"])


@when("j'envoie DELETE /booking/{id} avec un token invalide")
def delete_invalid_token(ctx):
    resp_create = ctx["booking"].create_booking()
    ctx["booking_id"] = resp_create.json()["bookingid"]
    ctx["response"] = BookingPage(token="FAKE_TOKEN_INVALID").delete_booking(
        ctx["booking_id"]
    )


@when("j'envoie DELETE /booking/9999999 avec mon token")
def delete_inexistant(ctx):
    ctx["response"] = ctx["booking"].delete_booking(9999999)


@when("j'envoie DELETE /booking/{id} une seconde fois")
def delete_booking_twice(ctx):
    # Crée une réservation, la supprime une première fois, puis une seconde
    resp = ctx["booking"].create_booking()
    bid  = resp.json()["bookingid"]
    ctx["booking"].delete_booking(bid)          # 1er DELETE → 201
    ctx["response"] = ctx["booking"].delete_booking(bid)  # 2e DELETE → 404/405


# ── Then : assertions DELETE ──────────────────────────────────────────────────
# Note: 'le body de la reponse contient "Created"' est dans common_steps.py

File: api-Java-Rest-Assured/src/test/java/com/restfulbooker/steps/BookingDeleteSteps.java
package com.restfulbooker.steps;

import com.restfulbooker.client.BookingClient;
import com.restfulbooker.context.ScenarioContext;
import io.cucumber.java.en.When;

/**
 * Steps BDD pour booking_delete.feature -- DELETE /booking/{id}.
 * Équivalent Java de features/steps/booking_delete_steps.py.
 */
public class BookingDeleteSteps {

    private final ScenarioContext ctx;

    public BookingDeleteSteps(ScenarioContext ctx) {
        this.ctx = ctx;
    }

    @When("^j'envoie DELETE /booking/\\{id\\} avec mon token \\(Cookie\\)$")
    public void deleteBookingValid() {
        ctx.setResponse(ctx.getBooking().deleteBooking(ctx.getBookingId()));
    }

    @When("^j'envoie DELETE /booking/\\{id\\} sans header d'authentification$")
    public void deleteNoToken() {
        var respCreate = ctx.getBooking().createBooking();
        ctx.setBookingId(respCreate.jsonPath().getInt("bookingid"));
        ctx.setResponse(new BookingClient().deleteBooking(ctx.getBookingId()));
    }

    @When("^j'envoie DELETE /booking/\\{id\\} avec un token invalide$")
    public void deleteInvalidToken() {
        var respCreate = ctx.getBooking().createBooking();
        ctx.setBookingId(respCreate.jsonPath().getInt("bookingid"));
        ctx.setResponse(new BookingClient("FAKE_TOKEN_INVALID").deleteBooking(ctx.getBookingId()));
    }

    @When("^j'envoie DELETE /booking/9999999 avec mon token$")
    public void deleteInexistant() {
        ctx.setResponse(ctx.getBooking().deleteBooking(9999999));
    }

    @When("^j'envoie DELETE /booking/\\{id\\} une seconde fois$")
    public void deleteBookingTwice() {
        var resp = ctx.getBooking().createBooking();
        int bid = resp.jsonPath().getInt("bookingid");
        ctx.getBooking().deleteBooking(bid);
        ctx.setResponse(ctx.getBooking().deleteBooking(bid));
    }
}
